GA Mp7.1

Yas-Franco opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

##########################################################################################
For issue #31673

Yas-Franco committed:
GA Mp7.1

Yas-Franco committed:
GA MpTelemetry2.1 EE10

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=c8559d5d-d721-47f2-98a0-817c4917f505) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Yas-Franco committed:
GA Mp7.1

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=da37bf04-f2fa-4cfa-8e27-0aab25d64e7c) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented on dev/build.image/profiles/microProfile7/features.xml:
```diff
@@ -1,4 +1,5 @@
 <feature>microProfile-7.0</feature>
+<feature>microProfile-7.1</feature>
```
For both the `microProfile7` and `jakartaee11` profiles you will now need to add or update a `packageServerConflict` property to their package tasks in `build.image/build.gradle` so that they package correctly now that they have conflicting MP features.  You can see an example in #26581, but will need to do the right conflicts for MP 7 and 7.1.

jhanders34 replied:
I did make the assumption that the EE 10 profiles were ok because we already have multiple versions of mpOpenAPI and mpTelemetry in them but with some of the new feature restructures, there may be some extras to add there as well.  You can validate that the are good by taking the EE 10 profile zips and trying to start MP 6, 6.1, 7.0 and 7.1 from the zip.  If they fail to start due to missing features, you will know there are some conflict features you need to add.

Yas-Franco committed:
Update build.image packageServerConflict properties

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f8fe530c-9459-4032-8bc6-7170123673be) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Yas-Franco committed:
Update build.image packageServerConflict properties

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f3db6bf8-78ef-4ac2-bded-980f8ae2fd07) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Yas-Franco committed:
Update build.image packageServerConflict properties

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=b910b77a-b1be-4060-ad6b-37e6ada3d75f) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Yas-Franco committed:
Update build.image packageServerConflict properties

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=412158dd-729b-434b-9b8b-ffa2b2326766) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented on dev/build.image/build.gradle:
```diff
@@ -585,7 +585,7 @@ if (isAutomatedBuild && !isIFIXBuild) {
         dependsOn ':com.ibm.websphere.appserver.features:publishFeatureResources'
         withFeatures this.&jakartaee11Features
-        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc"
+        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc,io.openliberty.microProfile,io.openliberty.mpConfig,io.openliberty.mpFaultTolerance,io.openliberty.mpOpenAPI,io.openliberty.mpRestClient,io.openliberty.mpMetrics,io.openliberty.mpTelemetry,io.openliberty.org.eclipse.microprofile.config,io.openliberty.org.eclipse.microprofile.faulttolerance,i ...
```
```suggestion
        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc,io.openliberty.microProfile,io.openliberty.mpOpenAPI,io.openliberty.mpTelemetry,io.openliberty.org.eclipse.microprofile.openapi,io.openliberty.mpCompatible,io.openliberty.internal.mpVersion"
```

jhanders34 commented on dev/build.image/build.gradle:
```diff
@@ -627,6 +627,7 @@ if (isAutomatedBuild && !isIFIXBuild) {
         dependsOn parent.subprojects.assemble
         dependsOn ':com.ibm.websphere.appserver.features:publishFeatureResources'
         withFeatures this.&microProfile7Features
+        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc,io.openliberty.microProfile,io.openliberty.mpConfig,io.openliberty.mpFaultTolerance,io.openliberty.mpOpenAPI,io.openliberty.mpRestClient,io.openliberty.mpMetrics,io.openliberty.mpTelemetry,io.openliberty.org.eclipse.microprofile.config,io.openliberty.org.eclipse.microprofile.faulttolerance,i ...
```
```suggestion
        packageServerConflict = "io.openliberty.microProfile,io.openliberty.mpOpenAPI,io.openliberty.mpTelemetry,io.openliberty.org.eclipse.microprofile.openapi,io.openliberty.mpCompatible,io.openliberty.internal.mpVersion"
```

jhanders34 reviewed (changes requested):
I believe these are the right conflicts for MP 7 and 7.1 like what you did for webprofile 11.

jhanders34 commented:
> I believe these are the right conflicts for MP 7 and 7.1 like what you did for webprofile 11.

I tried it out locally setting the properties in server.xml and calling package directly.  The microProfile7 zip works correctly.  The webprofile and jakartaee 11 ones do not, but that is ok likely because those are not currently shipped at all.  

The worse problem is that jakartaee10 and webprofile10 do not work and not is not the fault of this PR, but rather an issue introduced with the new mpHealth 4.0 working with older versions of MP.  It may also have not worked even before that changes since mpHealth 4.0 already supported multiple versions of MP.  Next I need to confirm if this has been broken since that change was added in.

I just tried it again and it is working.  Hmmm.

jhanders34 commented:
I found a way to get the EE 11 ones working, but again that can be done outside of this PR.  The MP 7 and EE 10 zips are the important ones that need to be right for this PR.

jhanders34 committed:
Updates to get microprofile7 zip to at least work

jhanders34 commented:
I applied the changes in order to get a build going and to get this moving along since we are running low on time.  I did additional testing with the EE 10 zip and it is messed up due to the mpHealth 4.0 changes and probably has been for quite some time.  I have an idea that can be used to fix both the ee10 and ee11 packages, but it will require updating some of the kernel code on how package server conflicts are handled.

!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=c6452048-a686-47a8-b53a-e80df6c104c1) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Yas-Franco committed:
Update build.image profiles for mp 7.1

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=2fc67a46-dca5-431d-af4b-d8dbad17e80a) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jhanders34 commented on dev/build.image/build.gradle:
```diff
@@ -566,7 +566,7 @@ if (isAutomatedBuild && !isIFIXBuild) {
         dependsOn ':com.ibm.websphere.appserver.features:publishFeatureResources'
         withFeatures this.&webProfile11Features
-        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc"
+        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc,io.openliberty.microProfile,io.openliberty.mpOpenAPI,io.openliberty.mpTelemetry,io.openliberty.org.eclipse.microprofile.openapi,io.openliberty.mpCompatible,io.openliberty.internal.mpVersion"
```
```suggestion
        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc"
```

jhanders34 commented on dev/build.image/build.gradle:
```diff
@@ -585,7 +585,7 @@ if (isAutomatedBuild && !isIFIXBuild) {
         dependsOn ':com.ibm.websphere.appserver.features:publishFeatureResources'
         withFeatures this.&jakartaee11Features
-        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc"
+        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc,io.openliberty.microProfile,io.openliberty.mpOpenAPI,io.openliberty.mpTelemetry,io.openliberty.org.eclipse.microprofile.openapi,io.openliberty.mpCompatible,io.openliberty.internal.mpVersion"
```
```suggestion
        packageServerConflict = "com.ibm.websphere.appserver.sessionStore,com.ibm.websphere.appserver.jdbc"
```

jhanders34 reviewed (changes requested):
Since no actual conflict now, you shouldn't have the conflicts listed

jhanders34 committed:
Do not set conflicts for MP features for EE 11 packages

Now that we only have 1 MP convenience feature in EE 11 packages, we do not need the conflict for MP feature any longer.

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=1d0102ce-b1ab-4375-b180-5efcfb3f3a98) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

Yas-Franco commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=de33cf22-d0bb-48f4-91a5-5ab6370eb608) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._
